idioms/priv-extend.md

```diff
@@ -39,8 +80,37 @@ The downside of this approach is that you might need to add an otherwise unneede
 field to the struct. You can use the `()` type so that there is no runtime overhead
 and prepend `_` to the field name to avoid the unused field warning.
 
-If Rust allowed private variants of enums, we could use the same trick to make
-adding a variant to an enum backwards compatible. The problem there is exhaustive
-match expressions. A private variant would force clients to have a `_` wildcard
-pattern. A common way to implement this instead is using the [`#[non_exhaustive]`](<https://doc.rust-lang.org/reference/attributes/type_system.html>)
-attribute.
+```rust
+pub struct S {
+    pub a: i32,
+    // Because `b` is private, you cannot match on `S` without using `..` and `S`
+    //  cannot be directly instantiated or matched against
+    _b: ()
+}
+```
+
+## Discussion
+
+On `struct`s `#[non_exhaustive]` allows adding additional fields in a backwards
+compatible way. It will also prevent clients from using the struct constructor,
+even if all the fields are public. This may be helpful, but it's worth considering
+if you _want_ an additional field to be found by clients as a compiler error rather
+than something that may be silently undiscovered.
+
+Finally, #[non_exhaustive] can be applied to enum variants. A `#[non_exhaustive]`
+variant behaves in the same way as a `#[non_exhaustive]` struct.
+
```

Comment:
```suggestion
```
I think this is just a repetition and this was already clear.

If we decide to keep it, let's add " ` " around #[non_exhaustive]